components/suggest/src/store.rs

```diff
@@ -4034,13 +4132,67 @@ mod tests {
 
         let store = unique_test_store(SnapshotSettingsClient::with_snapshot(snapshot));
         store.dbs()?.writer.write(|dao| {
-            dao.put_meta(LAST_INGEST_META_KEY, 30)?;
+            dao.put_meta("last_quicksuggest_ingest_data", 30)?;
             Ok(())
         })?;
         store.ingest(SuggestIngestionConstraints::default())?;
 
         store.dbs()?.reader.read(|dao| {
-            assert_eq!(dao.get_meta::<u64>(LAST_INGEST_META_KEY)?, Some(30));
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_data")?,
+                Some(30)
+            );
+            Ok(())
+        })?;
+
+        Ok(())
+    }
+
+    /// Tests that we only ingest providers that we're concerned with.
+    #[test]
+    fn ingest_constraints_provider() -> anyhow::Result<()> {
+        before_each();
+
+        let snapshot = Snapshot::with_records(json!([{
+            "id": "data-1",
+            "type": "data",
+            "last_modified": 15,
+        }, {
+            "id": "icon-1",
+            "type": "icon",
+            "last_modified": 30,
+        }]))?;
+
+        let store = unique_test_store(SnapshotSettingsClient::with_snapshot(snapshot));
+        store.dbs()?.writer.write(|dao| {
+            // Check that existing data is updated properly.
+            dao.put_meta("last_quicksuggest_ingest_data", 10)?;
+            Ok(())
+        })?;
+
+        let constraints = SuggestIngestionConstraints {
+            max_suggestions: Some(100),
+            providers: Some(vec![SuggestionProvider::Amp, SuggestionProvider::Pocket]),
+        };
+        store.ingest(constraints)?;
+
+        store.dbs()?.reader.read(|dao| {
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_data")?,
+                Some(15)
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_icon")?,
+                Some(30)
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_pocket-suggestions")?,
+                None
+            );
+            assert_eq!(
+                dao.get_meta::<u64>("last_quicksuggest_ingest_amo-suggestions")?,
+                None
+            );
             Ok(())
```

Comment:
I was originally not super happy with the tests because the snapshot returns the same data every time we call remote settings (in this case, twice). This means that although we're trying to get only "Pocket" data, the snapshot returns whatever we configured it to return (for all the other types as well). 

Now that I'm thinking of it, the core functionality of the test, which is to separate the last_ingest record, kind of _works_ as it does properly separate the values in storage properly. I don't know, what do you think @linabutler ?